packages/web/src/hooks/useErrorPageOnFailedStatus.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react'
+
+import { Status } from '@audius/common'
+
+import { ERROR_PAGE } from 'utils/route'
+
+import { useNavigateToPage } from './useNavigateToPage'
+
+/** Automatically navigates to the `/error` route if the provided status is
+ * `Status.ERROR`. NOTE: If the status passed to this hook is derived from an
+ * `audius-query` data fetching hook, the page using this hook needs to support
+ * resetting of the fetch state on mount (typically through the `force` option
+ * of `useQuery`). Otherwise, the user will be immediately redirected back to the
+ * error page if they hit 'Try Again'.
+ */
+export const useErrorPageOnFailedStatus = (status: Status) => {
+  const navigate = useNavigateToPage()
+  useEffect(() => {
+    if (status === Status.ERROR) {
+      navigate(ERROR_PAGE)
```

Comment:
If we're on board with toasts being the future of unhandled errors, or even a slight improvement over what we have today, I'd argue it's worth having this hook dispatch a toast action instead. Would avoid the entanglement of the `force` option being necessary to avoid endless retry loop

Reply:
I agree. We don't currently have any designs for error toasts. I can follow up and see if there is time/desire to do that right now. Otherwise, my plan was to leave this as is and then come back and update any components which are using this hook / strategy to follow whatever our updated error pattern is.